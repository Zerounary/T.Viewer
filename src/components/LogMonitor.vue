<template>
  <v-container>
    <v-row
      text-center
      wrap
    >
      <v-btn
        v-bind:outlined="!isListenerOn"
        block
        color="primary"
        v-on:click="onStartButton"
        >
        {{ btnText }}
      </v-btn>
      {{ message }}
    </v-row>
  </v-container>
</template>

<script>
import LogListener from '../LogListener';
import { ipcRenderer } from 'electron';

export default {
  data: () => ({
      message: 'No Log',
      isListenerOn: true,
      logListener: new LogListener(ipcRenderer),
  }),
  computed: {
    btnText() {
      return this.isListenerOn ? 'On' : 'Off';
    }
  },
  methods: {
    onStartButton: function () {
      if (this.isListenerOn) {
        this.isListenerOn = false;
        this.logListener.registerListener((msg) => {
          this.message = msg;
        });
      } else {
        this.isListenerOn = true;
        this.logListener.unregisterListener();
      }
    }
  }
};
</script>
