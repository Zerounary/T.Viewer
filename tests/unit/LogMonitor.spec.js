import LogMonitor from '@/components/LogMonitor.vue'
import Vue from 'vue';
import Vuetify from 'vuetify';

// Utilities
import {
  mount,
  createLocalVue
} from '@vue/test-utils'
Vue.use(Vuetify);

const localVue = createLocalVue()
describe('LogMonitor.vue', () => {
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  it('should have a default content and match snapshot', () => {
    const wrapper = mount(LogMonitor, {
      localVue,
      vuetify,
    })

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()
  })

  it('should call registerListener', () => {
    const vm = new Vue(LogMonitor).$mount();
    vm.isListenerOn = true;
    vm.logListener.registerListener = jest.fn();

    vm.onStartButton();
    expect(vm.logListener.registerListener).toBeCalledTimes(1);
  })

  it('should call unregisterListener', () => {
    const vm = new Vue(LogMonitor).$mount();
    vm.isListenerOn = false;
    vm.logListener.unregisterListener = jest.fn();

    vm.onStartButton();
    expect(vm.logListener.unregisterListener).toBeCalledTimes(1);
  })
})
