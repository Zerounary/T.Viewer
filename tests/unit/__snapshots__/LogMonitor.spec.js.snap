// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogMonitor.vue should have a default content and match snapshot 1`] = `
<div class="container">
  <div text-center="" wrap="" class="row"><button type="button" class="v-btn v-btn--block v-btn--contained theme--light v-size--default primary"><span class="v-btn__content">
      On
    </span></button>
    No Log
  </div>
</div>
`;
